Survey/Sources/Presentation/Modules/SurveyDetail/SurveyDetailView.swift

```diff
@@ -55,23 +57,37 @@ struct SurveyDetailView: View {
                 }
             }
         })
+        .onReceive(output.$survey) {
+            guard let questions = $0?.questions else { return }
+            surveyQuestions = questions
+            withoutAnimation {
+                self.willShowQuestions.send(!surveyQuestions.isEmpty)
+            }
+        }
         .onAppear(perform: {
             fadeInOut = false
             withAnimation(Animation.easeInOut(duration: 1.0)) {
                 fadeInOut.toggle()
             }
         })
-        .onReceive(output.$willGoToNextSurvey) {
-            guard $0 else { return }
-            withoutAnimation {
-                isSurveyQuestionPresented.toggle()
-            }
-        }
-        .fullScreenCover(isPresented: $isSurveyQuestionPresented) {
-            SurveyQuestionView(isPresented: $isSurveyQuestionPresented)
+        .fullScreenCover(isPresented: $output.isSurveyQuestionPresented) {
+            SurveyQuestionView(
+                isPresented: $output.isSurveyQuestionPresented,
```

Comment:
On `SurveyQuestionView` we are setting this output value explicitly, this could be a `PassthroughSubject` but I think that bring too much complexity to `SurveyQuestionView`.